libtsuba/src/RDGCore.cpp

```diff
@@ -26,11 +26,29 @@ AddProperties(
     int last = current->num_columns();
 
     for (int i = 0, n = schema->num_fields(); i < n; i++) {
-      auto result =
-          next->AddColumn(last + i, schema->field(i), props->column(i));
+      auto name = schema->field(i)->name();
+      auto current_col = next->GetColumnByName(name);
+      arrow::Result<std::shared_ptr<arrow::Table>> result;
+      std::string error_context = "insert";
+      if (current_col == nullptr) {
```

Comment:
Is this condition guaranteed to return the same value for all iterations of the for-loop?

Reply:
No.

Reply:
The schema is for the new properties to insert. I thought it was the old schema earlier. Makes sense now.